documentation/src/main/asciidoc/topics/con_estimating_size_data_set.adoc

```diff
@@ -0,0 +1,30 @@
+[id='data-set-size_{context}']
+= Estimating the size of your data set
+Calculate your total data set to work out the correct heap size per JVM.
+
+As a rough guide, you can estimate the total size of your data set with this formula:
+
+[source,options="nowrap",subs=attributes+]
+----
+Number of entries * (Key size + Value size + Memory overhead)
+----
+
+[NOTE]
+====
+With remote caches you need to calculate key sizes and value sizes in their marshalled forms.
+====
+
+.Memory overhead
+
+In the preceding formula, memory overhead refers to the additional memory that {brandname} uses to store entries.
+Overhead depends on whether you store data in the JVM heap or off-heap in native memory.
+
+For JVM heap, you should calculate approximately 200 bytes per entry as memory overhead.
+With off-heap storage, {brandname} uses 25 bytes without eviction or 61 bytes with eviction.
+
+In addition to memory overhead that {brandname} imposes per entry, you need to factor in other processes that increase overall memory usage.
+
+* Rebalancing operations for clusters where nodes join and leave.
```

Comment:
Here Heap is needed in any case even if if all cached date is off-heap
https://access.redhat.com/solutions/4343611